src/org/org.ts

```diff
@@ -542,6 +542,43 @@ export class Org extends AsyncOptionalCreatable<Org.Options> {
     }
   }
 
+  /**
+   * Returns `true` if the org uses source tracking.
+   * Side effect: updates files where the property doesn't currently exist
+   */
+  public async usesTracking(): Promise<boolean> {
+    // use the property if it exists
+    const tracking = this.getField(Org.Fields.TRACKING);
+    if (isBoolean(tracking)) {
+      return tracking;
+    }
+    // scratch orgs with no property use tracking by default
+    if (await this.determineIfScratch()) {
+      // save true for next time to avoid checking again
+      await this.setTracking(true);
+      return true;
+    }
+    if (await this.determineIfSandbox()) {
+      // does the sandbox know about the SourceMember object?
+      const supportsSourceMembers = await this.supportsSourceTracking();
+      await this.setTracking(supportsSourceMembers);
+      return supportsSourceMembers;
+    }
+    // any other non-sandbox, non-scratch orgs won't use tracking
+    await this.setTracking(false);
+    return false;
+  }
+
+  /**
+   * Set the tracking property on the org's auth file
+   *
+   * @param value true or false (whether the org should use source tracking or not)
+   */
+  public async setTracking(value: boolean): Promise<void> {
+    const originalAuth = await AuthInfo.create({ username: this.getUsername() });
+    originalAuth.save({ tracking: value });
```

Comment:
I suggest calling the authinfo handle side effects function instead if save. Ensures tracking is only set via side effects.

Reply:
+1